SampleOfferWithoutReward.sol

```diff
@@ -118,15 +124,21 @@ contract SampleOfferWithoutReward {
             isContractValid = false;
     }
 
-    function getDailyPayment() {
-        if (msg.sender != contractor)
-            throw;
-        uint amount = (now - dateOfSignature + 1 days) / (1 days) * dailyWithdrawLimit - paidOut;
+    // Withdraw to the Contractor.
+    //
+    // Withdraw the amount of money the Contractor has right to according to
+    // the current withdraw limit.
+    // Executing this function before the Offer is signed off by the Client
+    // makes no sense as this contract has no money.
+    function getDailyPayment() noEther {
+        uint timeSinceLastPayment = now - lastPayment;
+        // Calculate the amount using 1 second precision.
+        uint amount = (timeSinceLastPayment * dailyWithdrawLimit) / (1 days);
```

Comment:
you are calculating "seconds since last payment" time `dailyWithdrawLimit` . This is way too much.
It neds to be : `(timeSinceLastPayment  / (1 days)) * dailyWithdrawLimit; `